### PR TITLE
metahelm/errors: add json struct tags for ChartError

### PR DESCRIPTION
**Pull request by mikeykhalil**
* adds json struct tags to ChartError and FailedPod types
* adds go vet step to circle ci (which would help catch tags with bad syntax, among many other issues)

### CODE REVIEW

File: pkg/metahelm/error.go [hunk 1]
```diff
@@ -16,9 +16,12 @@ import (
...
-	Conditions                   []corev1.PodCondition
-	ContainerStatuses            []corev1.ContainerStatus
+	Name              string                   `json:"name"`
+	Phase             string                   `json:"phase"`
+	Message           string                   `json:"message"`
+	Reason            string                   `json:"reason"`
+	Conditions        []corev1.PodCondition    `json:"conditions"`
+	ContainerStatuses []corev1.ContainerStatus `json:"container_statuses"`
 	// Logs is a map of container name to raw log (stdout/stderr) output
 	Logs map[string][]byte
```
Comment (bkeroackdsc) [inline pkg/metahelm/error.go:26]: Did you intend to omit the tag for this field?

**Review by bkeroackdsc — CHANGES_REQUESTED** [review]

**Comment by mikeykhalil** [comment]
> ci/circleci Expected — Waiting for status to be reported

This status is unrelated to this change. Just something that needs to be updated in the repo's settings. 

https://support.circleci.com/hc/en-us/articles/360004346254-Workflow-status-checks-never-completes-because-of-ci-circleci-Waiting-for-status-to-be-reported

File: pkg/metahelm/error.go (hunk 1 above)
Comment (mikeykhalil) [reply pkg/metahelm/error.go]: Was unintentional. I focused too much on ensuring go vet would catch bad struct tags as opposed to making sure all the nested fields actually had struct tags 🤦 . Thanks for catching.

**Review by bkeroackdsc — APPROVED** [review]
LGTM. After merging, create a new GitHub release (0.6.2) and then you can update the version required by dependents (Acyl).

**Pull request merged**